### PR TITLE
fix(linux): improve BCP 47 canonicalization

### PR DESCRIPTION
**Pull request by ermshiperete**
This is based on #4425. Also a few additional bug fixes that showed up on the way.

### CODE REVIEW

File: linux/keyman-config/tests/test_bcp47tag.py [hunk 1]
```diff
@@ -82,6 +82,14 @@ def test_SetTag(self):
         # Verify
         self.assertEqual(Bcp47Tag('sr', 'Cyrl'), sut)
 
+    def test_SetTag_Update(self):
+        # Setup
+        sut = Bcp47Tag.create('glr-Latn-LR')
+        # Execute
+        sut.tag = 'glr'
```
Comment (mcdurdin) [inline linux/keyman-config/tests/test_bcp47tag.py:89]: I'm not entirely sure what this is testing? How does it differ from the previous test?

**Review by mcdurdin — APPROVED** [review]
LGTM

File: linux/keyman-config/tests/test_bcp47tag.py (hunk 1 above)
Comment (ermshiperete) [reply linux/keyman-config/tests/test_bcp47tag.py:89]: You're right, that's confusing. I'll change it. It's testing that setting a new tag will clear out the unspecified parts.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.240-beta](https://keyman.com/downloads/releases/beta/14.0.240)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.16-alpha](https://keyman.com/downloads/releases/alpha/15.0.16)

**Pull request merged**